src/java/org/apache/cassandra/cql3/statements/SelectStatement.java

```diff
@@ -1458,7 +1458,8 @@ private void checkNeedsFiltering(StatementRestrictions restrictions) throws Inva
                 // We will potentially filter data if either:
                 //  - Have more than one IndexExpression
                 //  - Have no index expression and the row filter is not the identity
-                checkFalse(restrictions.needFiltering(), StatementRestrictions.REQUIRES_ALLOW_FILTERING_MESSAGE);
+                if (restrictions.requiresAllowFilteringIfNotSpecified())
```

Comment:
Move the condition to the if-statement above maybe? No need for extra nesting here.

Reply:
yeah I was thinking about moving it there but I like how `restrictions.needFiltering()` is isolated in what it does.